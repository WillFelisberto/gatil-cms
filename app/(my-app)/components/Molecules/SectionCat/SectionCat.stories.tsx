import type { Meta, StoryObj } from '@storybook/react';
import Image from 'next/image';

import { SectionCat } from './SectionCat';

const meta: Meta<typeof SectionCat> = {
  title: 'Molecules/SectionCat',
  component: SectionCat,
  tags: ['autodocs']
};

export default meta;
type Story = StoryObj<typeof SectionCat>;

const Content = ({ dark }: { dark?: boolean }) => (
  <div className="h-full flex flex-col md:flex-row items-center justify-center gap-8 px-4">
    {/* Texto */}
    <div className="md:w-1/2 text-center md:text-left">
      <h2 className="text-3xl md:text-4xl font-bold mb-6">Conheça nossos gatinhos resgatados</h2>
      <p className="text-lg md:text-xl leading-relaxed mb-8">
        Ao apadrinhar, você contribui diretamente para os cuidados diários, tratamentos e bem-estar
        dos nossos pequenos heróis. Clique abaixo e escolha o seu novo afilhado.
      </p>
      <button
        className={`px-8 py-3 rounded-full text-lg font-semibold transition-colors ${
          dark
            ? 'bg-[#A1DFFF] text-blue-950 hover:bg-[#7cd0f3]'
            : 'bg-blue-950 text-white hover:bg-[#001f4c]'
        }`}
      >
        Apadrinho M
      </button>
    </div>

    {/* Imagem */}
    <div className="md:w-1/2 flex justify-center">
      <div className="relative w-full h-64 md:h-96">
        <Image
          src="/gato.jpg"
          alt="Gato para adoção"
          fill
          className="rounded-xl shadow-lg object-cover"
        />
      </div>
    </div>
  </div>
);

export const Claro: Story = {
  args: {
    dark: false,
    children: <Content dark={false} />
  },
  parameters: {
    layout: 'fullscreen',
    backgrounds: {
      default: 'dark'
    }
  }
};

export const Escuro: Story = {
  args: {
    dark: true,
    children: <Content dark={true} />
  },
  parameters: {
    layout: 'fullscreen'
  }
};
