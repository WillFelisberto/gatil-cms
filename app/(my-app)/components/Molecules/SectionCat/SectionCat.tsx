'use client';

import { useIsMobile } from 'app/(my-app)/hooks/useIsMobile';
import type { ReactNode } from 'react';

const footerClaro = '/gatoazulrodape.webp';
const footerEscuro = '/gatobrancorodape.webp';

interface SectionCatProps {
  dark?: boolean;
  children?: ReactNode;
  label?: string;
}

export const SectionCat = ({ dark = false, children, label }: SectionCatProps) => {
  const isMobile = useIsMobile();
  const backgroundImage = dark ? footerEscuro : footerClaro;

  return (
    <section
      id="adoptable-cats-section"
      role="region"
      aria-label={label || 'Seção com informações sobre gatos disponíveis para adoção'}
      data-testid="section-cat-background"
      className={`
        relative w-full
        flex items-center
        overflow-hidden
        bg-bottom bg-no-repeat bg-contain
        ${dark ? 'bg-[#013274] text-white' : 'bg-white text-[#013274]'}
      `}
      style={{
        backgroundImage: `url(${backgroundImage})`,
        backgroundRepeat: 'no-repeat',
        backgroundPosition: 'bottom center',
        backgroundSize: 'inherit',
        height: isMobile ? '100vh' : '80vh'
      }}
    >
      <div className="w-full max-w-6xl mx-auto px-6 py-12">{children}</div>
    </section>
  );
};
