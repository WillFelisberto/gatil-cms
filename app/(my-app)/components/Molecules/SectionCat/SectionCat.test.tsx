import { render, screen } from '@/tests/test-utils';

import { SectionCat } from './SectionCat';

describe('<SectionCat />', () => {
  it('Should renders correctly with default label and light theme', () => {
    const { container } = render(
      <SectionCat>
        <p>Meow content</p>
      </SectionCat>
    );

    const section = screen.getByTestId('section-cat-background');
    expect(section).toBeInTheDocument();
    expect(section).toHaveAttribute(
      'aria-label',
      'Seção com informações sobre gatos disponíveis para adoção'
    );
    expect(screen.getByText('Meow content')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('Should applies dark theme when dark prop is true', () => {
    render(
      <SectionCat dark>
        <p>Dark cat</p>
      </SectionCat>
    );

    const section = screen.getByTestId('section-cat-background');
    expect(section.className.replace(/\s+/g, ' ')).toMatch(/bg-\[#013274\]/);
    expect(screen.getByText('Dark cat')).toBeInTheDocument();
  });

  it('Should uses custom aria-label when provided', () => {
    render(
      <SectionCat label="Custom label for SEO">
        <span>Custom Label Content</span>
      </SectionCat>
    );

    const section = screen.getByTestId('section-cat-background');
    expect(section).toHaveAttribute('aria-label', 'Custom label for SEO');
    expect(screen.getByText('Custom Label Content')).toBeInTheDocument();
  });
});
