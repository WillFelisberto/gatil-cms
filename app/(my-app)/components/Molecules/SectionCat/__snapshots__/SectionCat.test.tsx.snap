// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionCat /> Should renders correctly with default label and light theme 1`] = `
<div>
  <section
    aria-label="Seção com informações sobre gatos disponíveis para adoção"
    class="
        relative w-full
        flex items-center
        overflow-hidden
        bg-bottom bg-no-repeat bg-contain
        bg-white text-blue-950
      "
    data-testid="section-cat-background"
    id="adoptable-cats-section"
    role="region"
    style="background-image: url(/gatoazulrodape.webp); background-repeat: no-repeat; background-position: bottom center; background-size: inherit; height: 64vh;"
  >
    <div
      class="w-full max-w-6xl mx-auto px-6 py-12"
    >
      <p>
        Meow content
      </p>
    </div>
  </section>
</div>
`;
